File: IntentActivity/app/src/main/java/com/example/intentactivity/MainActivity.java
package com.example.intentactivity;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    Button btnNextActivity; //Cammel case letter
    EditText editText;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnNextActivity=findViewById(R.id.btn_next);
        editText=findViewById(R.id.edit_name);



        btnNextActivity.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                final String name=editText.getText().toString();
                Intent intent=new Intent(MainActivity.this,NextActivity.class);
                intent.putExtra("keyName",name);
                startActivity(intent);

            }
        });
    }
}

File: RetrofitPj/app/src/main/java/com/example/retrofitpj/MainActivity.kt
package com.example.retrofitpj

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var edt_name=findViewById<EditText>(R.id.edt)
        var btn_add=findViewById<Button>(R.id.btn_add)

        btn_add.setOnClickListener {
            var txt=edt_name.text

            var intent= Intent(this,SecondActivity::class.java)

            intent.putExtra("name",txt)
            intent.putExtra("Id",txt)
            startActivity(intent)

        }

    }
}
